File: BookShop.BusinessLogic/Models/DiscountModels/DiscountCreationModel.cs
using BookShop.BusinessLogic.PrintedEditionModels;
using System.Collections.Generic;

namespace BookShop.BusinessLogic.Models.DiscountModels
{
    public class DiscountCreationModel
    {
        public DiscountModel DiscountModel { get; set; }
        public List<PrintedEditionModel> PrintedEditionModels { get; set; }

        public DiscountCreationModel()
        {
            PrintedEditionModels = new List<PrintedEditionModel>();
        }
    }
}

File: BookShop.Presentation/wwwroot/ClientApp/src/app/models/discountModels/discountCreationModel.ts
import { DiscountModel } from 'src/app/models/discountModels/discountModel';
import { PrintedEditionModel } from 'src/app/models/printedEditionModels/printedEditionModel';

export class DiscountCreationModel {
    public discountModel: DiscountModel;
    public printedEditionModels: PrintedEditionModel[];

    constructor() {
        this.initializeDefaultModel();
    }

    private initializeDefaultModel() {
        this.discountModel = new DiscountModel();
        this.printedEditionModels = [];
    }
}